analyzers/ntuple.py

```diff
@@ -74,10 +79,35 @@ def bookJet( tree, pName ):
     bookP4(tree, pName )
     for pdgid in pdgids:
         bookComponent(tree, '{pName}_{pdgid:d}'.format(pName=pName, pdgid=pdgid))
-    # var(tree, '{pName}_npart'.format(pName=pName))
+    var(tree, '{pName}_TCHP'.format(pName=pName))
+    var(tree, '{pName}_TCHP_IP'.format(pName=pName))
+    for tagger in ['IP_b_LL', 'IPs_b_LL', 'TCHE', 'TCHE_IP', 'TCHE_x', 'TCHE_y', 'TCHE_z', 'TCHE_xy', 'TCHE_pt', 'TCHE_dr']:
+        var(tree, '{pName}_{tagger}'.format(pName=pName, tagger=tagger))
+    for ptc in ['K0s', 'Kp', 'L0', 'S0', 'Sp', 'Sm', 'Muons'] :
+        var(tree, '{pName}_{ptc}'.format(pName=pName, ptc=ptc))
 
 def fillJet( tree, pName, jet ):
     fillP4(tree, pName, jet )
+    for tagger in ['IP_b_LL', 'IPs_b_LL', 'TCHE', 'TCHE_IP', 'TCHE_x', 'TCHE_y', 'TCHE_z', 'TCHE_xy', 'TCHE_pt', 'TCHE_dr']:
+        if jet.tags.get(tagger, None) is not None:
```

Comment:
Ah of course, thanks.

then to check for the existence of the tagger and get the value, you could do: 

    if tagger in jet.tags: 
         value = jet.tags[tagger]


Le 30 juin 2016 à 13:02, lucastorterotot <notifications@github.com> a écrit :

> In analyzers/ntuple.py:
> 
> >  
> >  def fillJet( tree, pName, jet ):
> >      fillP4(tree, pName, jet )
> > +    for tagger in ['IP_b_LL', 'IPs_b_LL', 'TCHE', 'TCHE_IP', 'TCHE_x', 'TCHE_y', 'TCHE_z', 'TCHE_xy', 'TCHE_pt', 'TCHE_dr']:
> > +        if jet.tags.get(tagger, None) is not None:
> Your suggestion works when no value of the tagger is equal to 0. If so, 0 is considered a boolean and the IF condition is False. But it only happend when the value is exactly 0.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>